apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,206 @@
+    fontFamily: "'Gotham 5r', sans-serif",
+  },
+  middle: {
+    marginTop: 20,
+    marginBottom: 20,
+    paddingBottom: 20,
+    paddingTop: 20,
+    borderTopWidth: 1,
+    borderBottomWidth: 1,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    borderStyle: 'solid',
+    borderColor: color.lighter_gray,
+  },
+  bottom: {
+    display: 'flex',
+    justifyContent: 'flex-end',
+  },
+  error: {
+    color: color.red,
+  },
+};
+
+export default class SchoolInfoInterstitial extends React.Component {
+  static propTypes = {
+    // This component is tightly bound to the HAML view that renders it and
+    // populates its props, and similarly to the User update API that
+    // it uses to save entered school info.
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        // Note, these names intentionally match the fields on the SchoolInfo
+        // model, not their JavaScript-friendly equivalents.  The mapping
+        // occurs in the constructor, below, and back when we submit.
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        state: PropTypes.string,
+        zip: PropTypes.string,
+        full_address: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    onClose: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    onClose: function () {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    const {existingSchoolInfo} = this.props.scriptData;
+    const initialNcesSchoolId = existingSchoolInfo.school_id ?
+      existingSchoolInfo.school_id :
+      (
+        existingSchoolInfo.country === 'United States'
+        &&
+        SCHOOL_TYPES_HAVING_NCES_SEARCH.includes(existingSchoolInfo.school_type)
+        &&
+        (existingSchoolInfo.school_name || existingSchoolInfo.state || existingSchoolInfo.zip)
+      ) ? '-1' : '';
+
+    this.state = {
+      country: existingSchoolInfo.country || '',
+      schoolType: existingSchoolInfo.school_type || '',
+      schoolName: existingSchoolInfo.school_name || '',
+      schoolState: existingSchoolInfo.state || '',
+      schoolZip: existingSchoolInfo.zip || '',
+      schoolLocation: existingSchoolInfo.full_address || '',
+      ncesSchoolId: initialNcesSchoolId,
+      showSchoolInfoUnknownError: false,
+    };
+  }
+
+
+  handleSchoolInfoSubmit = () => {
+    let schoolData;
+    if (this.state.ncesSchoolId === '-1') {
+      schoolData = {
+        "user[school_info_attributes][school_name]": this.state.schoolName,
+        "user[school_info_attributes][school_state]": this.state.schoolState,
+        "user[school_info_attributes][school_zip]": this.state.schoolZip,
+        "user[school_info_attributes][full_address]": this.state.schoolLocation,
+      };
```

Comment:
It doesn't look like `this.state.schoolLocation` ever gets updated from the original value. Since you aren't using the google location search this should probably get set to null here.

Reply:
I actually don't know whether we want the google location search or not - I'll check with Poorva.  What's the potential harm in keeping this wired up, to minimize the work to enable it later?

Reply:
In the case that the user entered new info, you will be keeping the old `full_address` resulting in a mashup of the old and new data. If they are submitting the form then the old data ought to be cleared out.

Reply:
Oh - the guidance I got was that this dialog should be like the school info on the signup page as much as possible, and that one uses Google Location search.  So instead of removing this I'm going to enable that feature.

Reply:
Keep in mind the note [here](https://github.com/code-dot-org/code-dot-org/blob/c417909cd16f7b062cf86b83df25f75ed40e35e3/apps/src/templates/SchoolNotFound.jsx#L69) about including the google maps script.